libexec/python/shub/api.py

```diff
@@ -211,22 +211,43 @@ def get_image_name(manifest, extension="simg"):
         default_naming = False
 
     # Second preference goes to commit
-    elif SHUB_NAMEBYCOMMIT is not None and manifest['version'] is not None:
-        image_name = "%s.%s" % (manifest['version'], extension)
+    elif SHUB_NAMEBYCOMMIT is not None:
+
+        if manifest.get("commit") is not None:
+            commit = manifest['commit']
+
+        elif manifest['version'] is not None:
```

Comment:
Having `elif` instead of `if` suggests that if `commit` is a thing `version` will not be a thing.  Or it will be ignored.  Is that accurate?

Reply:
Yes, first priority goes to the key "commit" in the manifest, over "version." The reason version is there is that for old singularity hub (becoming deprecated) the version did hold the commit. The entire second part there will likely go away after old Singularity Hub is turned off.